### DISCUSSION TITLE
#13 Wallet mining: get all the tokens in the walletDeployer contract without using the deployer and wards account?

### DISCUSSION THREAD

**Discussion by naiba** [tinchoabbate/damn-vulnerable-defi#24] [Help / Q&A / Troubleshooting] [answered]
Hi there, Just want to confirm if there really is a way to get all the tokens in the walletDeploy contract without using the deployer and wards account?

(Can you give me a hint?

**Comment by tinchoabbate** [comment, marked as answer]
The challenge can be solved using just the `player` account.

Also, the scenario presented in the challenge is quite similar to an issue that happened not so long ago. My impression is that with some research it should be feasible.

**Hint:** Start by learning how Gnosis factories and master copies are deployed deterministically across chains.

**Reply by naiba** [reply]
Yes, I know what happened on the optimistic chain, I have redeployed and got 20 million dvt, but it seems there is no way to bypass the can check to get all 43 dvt of deployment rewards. Can you give a little hint? This is my last remaining question in this challenge, thank you very much.

**Reply by tinchoabbate** [reply]
Ok, so you're almost there. Now you might want to take a closer look at the `AuthorizerUpgradeable` contract.

**Reply by naiba** [reply]
I've spent three nights checking the "can" method and the "WalletDeployer" contract, and its implementation is so simple that I can't find out the point with it.

Can you please check if your test case misuses an account other than the player account?

**Reply by naiba** [reply]
Could you please check if your test case is using ".connect(player)" with "authorizer"?
Sorry to bother you so much, cause I feel very confused.

**Reply by naiba** [reply]
Hi, @tinchoabbate, I have been looking at `AuthorizerUpgradeable` for a long time and have tried a lot but still haven't found the way, can you give me more detailed hints?

**Reply by naiba** [reply]
These are my efforts:

https://github.com/naiba-archived/damn-vulnerable-defi/blob/v3-solutions-by-naiba/test/wallet-mining/wallet-mining.challenge.js

https://github.com/naiba-archived/damn-vulnerable-defi/tree/v3-solutions-by-naiba/contracts/wallet-mining/

**Comment by naiba** [comment]
Can anyone share a solution? I really can't find a way to get the 43 DVT on the `walletDeployer` contract using only the `player` account.

**Reply by naiba** [reply]
take a closer look at the AuthorizerUpgradeable contract you will do this.

**Reply by akshay-ap** [reply]
hey @naiba , were you able to solve it? Even I have been trying solve the part where WalletDeployer tokens are transferred to player but, no success so far.

**Reply by naiba** [reply]
keep it up.
